src/saltext/vmware/states/vmc_security_groups.py

```diff
@@ -0,0 +1,421 @@
+                )
+
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Updated Security group {}".format(security_group_id),
+                old_state=existing_security_group,
+                new_state=get_security_group_after_update,
+                result=True,
+            )
+        else:
+            log.info("All fields are same as existing Security group %s", security_group_id)
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Security group exists already, no action to perform",
+                result=True,
+            )
+    else:
+        log.info("No Security group found with Id %s", security_group_id)
+        created_security_group = __salt__["vmc_security_groups.create"](
+            hostname=hostname,
+            refresh_key=refresh_key,
+            authorization_host=authorization_host,
+            org_id=org_id,
+            sddc_id=sddc_id,
+            domain_id=domain_id,
+            security_group_id=security_group_id,
+            verify_ssl=verify_ssl,
+            cert=cert,
+            expression=expression,
+            description=description,
+            tags=tags,
+        )
+
+        if "error" in created_security_group:
+            return vmc_state._create_state_response(
+                name=name, comment=created_security_group["error"], result=False
+            )
+
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Created Security group {}".format(security_group_id),
+            new_state=created_security_group,
+            result=True,
+        )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    domain_id,
+    security_group_id,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given security group does not exist on given SDDC
+
+    hostname
+        The host name of NSX-T manager
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console
+
+    org_id
+        The Id of organization to which the SDDC belongs to
+
+    sddc_id
+        The Id of SDDC from which the security group should be deleted
+
+    domain_id
+        The domain_id for which the security group should belong to. Possible values: mgw, cgw
+
+    security_group_id
+        Id of the security_group to be deleted from SDDC
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    """
+
+    log.info("Checking if Security group with Id %s is present", security_group_id)
```

Comment:
**suggestion** ID should be capitalized in docs prose and logging

Reply:
updated